packages/flutter_tools/lib/src/android/android_device.dart

```diff
@@ -652,6 +652,8 @@ class AndroidDevice extends Device {
         ...<String>['--ez', 'endless-trace-buffer', 'true'],
```

Comment:
there is an android unit test for passing debugging options, that should be expanded to cover this flag

Reply:
Done.